### DISCUSSION TITLE
Pybricks 3.3 Release Candidate out now. Please help us test!

### DISCUSSION THREAD

**Discussion by laurensvalk** [pybricks/support#1294] [Announcements]
We are super excited that we are finally getting ready for the official Pybricks 3.3 release!

That's right, no more *beta*, but a solid and stable release.

The major updates in this release are:
- Hub to hub communication
- Gyro support with DriveBase integration
- Cooperative multitasking
- Major motor and sensor accuracy updates

## We need your help!

How, you ask? Well:

# :arrow_right: :arrow_right:  **[Start coding, find bugs :bug: ! ](https://www.beta.pybricks.com/)** :arrow_left: :arrow_left: 

If you want to help, here's a few things to look out for especially:

- **Try the new features**
    - Multitask, broadcast, all the while driving in a precise curve! 
- **Firmware installation issues:**
   - Try it on Windows, Mac, or Linux
   - Try it on your latest Laptop and on your grandma's PC.
   - Try it with Google Chrome and Microsoft Edge.
   - Try including a program in the firmware via the settings menu.
- **Running program issues:**
   - Try super short programs or really large ones.
   - Try programs that print a lot, or nothing at all.
   - Try programs that use a lot of memory.
   - Try programs that run for a very long time.
   - Try to start and stop programs often.
   - Try to disconnect from one computer, then continue coding on another.
- **Hub, motor, and sensor issues:**
   - Try as many sensors, motors, and hubs as you have.
   - Try using lots of motors and sensors at the same time.
   - Try doing a lot with motors and sensors in fast loops.
   - Try low batteries and full batteries.
- **Documentation issues:**
   - Try as many of the included [example snippets](https://www.code.pybricks.com/) as you can.
   - Please stick to the documentation inside the coding app for now. While there's more on our [docs site](https://docs.pybricks.com/), that's reserved for a future release. Thanks!
   - Please share (or fix!) any typo or broken link that you can find.
- **Issues with the Pybricks Code editor:**
   - Try all the buttons, bells, and whistles.
   - Try the REPL for interactive coding.
   - Try saving and loading code files from your computer.
   - Try it in a website tab, and try installing it as an app via the settings menu.

Be sure to get creative and try to break as much as you can.

We'll try to release this version this week.

## Please tell us about your findings!

Please let us know about any errors or unexpected results. That way we can fix them before the final release.

Feel free to just comment here, even if everything is just working as expected. We'd love to hear that too.

You may also open a new [issue](https://github.com/pybricks/support/issues) directly.

## A big thanks to our users, contributors and sponsors!

We want to say *THANK YOU* to all our users, contributors, and [sponsors](https://github.com/sponsors/pybricks). You've been a great help!

**Comment by BertLindeman** [comment]
No more beta (YESSS) 

But for the Release candidate we still use beta.pybricks.com
Right?

**Reply by laurensvalk** [reply]
Yes, the link is in the post :smile: 

> No more beta (YESSS)

Wait until you find out we have been quietly working on a massive new beta, just around the corner :smile: Just needed to finish this one first!

**Reply by BertLindeman** [reply]
Probably my (old) eyes, I do not see the link.

**Reply by dlech** [reply]
It doesn't quite look like a normal link, but it is hiding in plain sight :smile: 


![image](https://github.com/pybricks/support/assets/963645/638a45fa-10c4-40f6-ba84-6f18d086f742)

**Comment by BertLindeman** [comment]
Did some tests, Windows11 / Linux Mint.
Spike prime / Technic hub.
All OK.
Have a send / observe test running from a RobotInventor hub to:
- Spike prime
- move hub
- city hub
- two technic hubs

All I ran into was sometimes an "unknown error" "Gatt error unknown" or Gatt connect failed"
1. on a connect to a hub
2. once on a firmware install,
3. a Technic hub (only that one) did not start a program; BlueTooth chip version v3.02.00

A simple repeat was enough to go on. Might well be the bluetooth activities on the observe tests.

a happy Pybricks user 

Thank you!

Bert

**Reply by laurensvalk** [reply]
Thanks for testing it on 6 hub Bert!

I think this is starting to look pretty good and we can do a proper release very soon now.

**Reply by laurensvalk** [reply]
@BertLindeman - can you try going to:

https://code.pybricks.com

Without doing any manual refreshing, what do you get?

A white screen or does it update normally? (saying a new version is available?). 

Same question for:

https://beta.pybricks.com

Bonus points for chrome and edge. Thanks!

**Reply by BertLindeman** [reply]
Was out volunteering in the library . . . 

All win11


First code.pybricks:
firefox:
![afbeelding](https://github.com/pybricks/support/assets/8142081/cd10b394-1e06-4747-ac4e-e9a50652b9e1)

Edge: White screen

Chrome I do not know if I refreshed already this morning and have a session open.
On new tab:
![afbeelding](https://github.com/pybricks/support/assets/8142081/313f3673-0ef5-4ed1-a061-ea280e9ed160)

~~No "new version" message~~ New version message OK.

Then beta.pybricks:
chrome:
It takes some time, then "new version" message.

Edge:
White screen

Did not do any refresh on them yet.

Bert

**Reply by dlech** [reply]
upgrading both `code.pybricks.com` and `beta.pybricks.com` worked correctly for me on Chrome on macOS (no white screen).

**Reply by BertLindeman** [reply]
Using NoScript extension on Edge.
After adding pybricks.com as allowed, no more problem on Edge, both beta and code.
![afbeelding](https://github.com/pybricks/support/assets/8142081/af81576d-8e60-4312-b23a-a8c4a11c311f)

**Reply by dlech** [reply]
I also had successful upgrade on Win11 + Edge (no white screen)

**Reply by laurensvalk** [reply]
Thank you both a lot for trying. That sounds pretty good.

@BertLindeman - for the cases where you saw a white screen, did closing the tab/browser and trying again at least reveal the new version?

The white screen sometimes seems to just happen. In my case, it happened on the "app" version on Android (which normally does support smooth updates as normal), so it's probably not tied to a particular browser but other auxiliary conditions.

But if it is recoverable by closing tabs/browser, that's not the worst. (It would be worse if clearing/resetting any caches was required).

Still, I suppose we could add some very basic content in the html placeholder template that tells people to close the tab/browser if this happens.

**Reply by BertLindeman** [reply]
> @BertLindeman - for the cases where you saw a white screen, did closing the tab/browser and trying again at least reveal the new version?
> 
> The white screen sometimes seems to just happen. In my case, it happened on the "app" version on Android (which normally does support smooth updates as normal), so it's probably not tied to a particular browser but other auxiliary conditions.

In my case while in the beta / code tab I enabled pybricks.com in NoScript.
Doing that does a refresh of the webpage and that got the page up OK.
